src/main/java/PageWithBase.java

```diff
@@ -0,0 +1,29 @@
+public abstract class PageWithBase {
+    private SFHeader header = new SFHeader();
+    private SFFooter footer = new SFFooter();
+
+
+    public SFHeader getHeader() {
+        return header;
+    }
+
+    public SFFooter getFooter() {
+        return footer;
+    }
+
+    private String email = "email";
+    private String password = "password";
+
+    public String getEmail() {
+        return email;
+    }
+
+    public SFHomePage signIN() {
+        var homePage = new SFHomePage()
+        .getHeader()
+                .openSignInPage()
+                .signIn(email, password);
+        return new SFHomePage();
```

Comment:
according to PO logic, you should not create a new PO object explicitly, you should get it from a method